src/SubtitlesParser/index.js

```diff
@@ -0,0 +1,189 @@
+/*
+ * If not stated otherwise in this file or this component's LICENSE file the
+ * following copyright and licenses apply:
+ *
+ * Copyright 2020 Metrological
+ *
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Log } from '../LightningSdkPlugins'
+
+export default class SubtitlesParser {
+  // @ params url: subtitle file URL
+  // @ customParser: a customParser to use instead of default parser of the plugin
+  // @ parseOptions.removeSubtitleTextStyles: remove subtitle textstyles possible value true or false
+  // @return parsed subtitles as list of objects
+  // also stores parsed data
+  static fetchAndParseSubs(
+    url,
+    customParser = false,
+    parseOptions = { removeSubtitleTextStyles: true }
+  ) {
+    const _url = new URL(url)
+    if (!((_url.protocol === 'https:' || _url.protocol === 'http:') && _url.hostname)) {
+      Log.info('Invalid subtitle Url')
+      return Promise.reject(new Error('Invalid URL'))
+    }
+
+    return new Promise((resolve, reject) => {
+      fetch(url)
+        .then(data => {
+          let subtitleData = data.text()
+
+          this.clearAllSubtitles()
+          if (customParser && typeof customParser === 'function') {
+            this._captions = customParser(subtitleData)
+          } else {
+            this._captions = this.parseSubtitles(subtitleData, parseOptions)
+          }
+          if (this._captions && this._captions.length) {
+            resolve(this._captions)
+          } else {
+            reject('Failed to parse subtitles: invalid subtitles length')
+          }
+        })
+        .catch(error => {
+          Log.error('Fetching subtitles file Failed:', error)
+          this.clearAllSubtitles()
+          reject('Fetching subtitles file Failed')
+        })
+    })
+  }
+
+  // clears stored subtitles data
+  static clearAllSubtitles() {
+    this._captions = null
+  }
+
+  // get current subtitles
+  // @ currentTime: currentTime in seconds
+  // @return: subtitle text at that currentTime
+  static getSubtitleByTimeIndex(currentTime) {
+    if (currentTime === undefined || isNaN(currentTime) || typeof currentTime !== 'number') {
```

Comment:
You can change this to: 
`if (typeof currentTime !== 'number' || currentTime === Infinity)`

`undefined` has a typeof `undefined`, so that covers your use case. `isNaN` and `typeof` kind of do the same thing, so it's a bit confusing to use both.